### ISSUE TITLE
[V4] Styles not applying at first build, but applying the new style or reopening the app

### ISSUE THREAD

**Issue by denysoleksiienko** [nativewind/nativewind#972]
**Describe the bug**
I using `nx monorepo` with `react-native CLI app`. I configured the `nativewind v4` with `tailwindcss 3.4.4`.
At first build the app, styles are default from ios. But when add new style, then is fine and all styles are applying from config

Looks like not creating .cache folder in node modules. Only after adding a new style. And I don't know, why `.cache/nativewind/global.css.ios.css` creating inside `apps/mobile` instead of root `node_modules`

```
workspace/
├── node_modules/
├── apps/
│   ├── mobile/
│   │   ├── node_modules/
│   │   │   ├── .cache/nativewind/global.css.ios.css
│   │   ├── android/
│   │   │   ├── app/
│   │   ├── ios/
│   │   │   ├── Podfile
│   │   │  

```
**To Reproduce**
```
tailwind.config.js

const { platformSelect } = require('nativewind/theme');

/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ['./src/App.{js,jsx,ts,tsx}', './src/**/*.{js,jsx,ts,tsx}'],
  presets: [require('nativewind/preset')],
  theme: {
    colors: {
      accent: {
        5: '#F3FCF7',
        10: '#E6FAF0',
        20: '#CDF5E1',
        30: '#B4EFD1',
        40: '#9BEAC2',
        60: '#6AE0A4',
        80: '#38D585',
        110: '#05B75D',
        120: '#05A252',
      },
    },
    fontFamily: {
      mulish: ['Mulish-Regular', 'sans-serif'],
      'mulish-bold': ['Mulish-Bold', 'sans-serif'],
      'mulish-italic': ['Mulish-Italic', 'sans-serif'],
      system: platformSelect({
        ios: 'San Francisco',
        android: 'Roboto',
        default: 'Mulish-Regular',
      }),
    },
  },
  plugins: [],
};
```

```
.babelrc.js

module.exports = function (api) {
  api.cache(true);

  if (
    process.env.NX_TASK_TARGET_TARGET === 'build' ||
    process.env.NX_TASK_TARGET_TARGET.includes('storybook')
  ) {
    return {
      presets: [
        [
          '@nx/react/babel',
          {
            runtime: 'automatic',
          },
        ],
      ],
    };
  }

  return {
    presets: ['module:@react-native/babel-preset', 'nativewind/babel'],
    plugins: ['react-native-reanimated/plugin'],
  };
};
```

```
metro.config.js

const { withNxMetro } = require('@nx/react-native');
const { getDefaultConfig, mergeConfig } = require('@react-native/metro-config');
const { withNativeWind } = require('nativewind/metro');

const defaultConfig = getDefaultConfig(__dirname);
const { assetExts, sourceExts } = defaultConfig.resolver;

/**
 * Metro configuration
 * https://reactnative.dev/docs/metro
 *
 * @type {import('metro-config').MetroConfig}
 */
const customConfig = {
  transformer: {
    babelTransformerPath: require.resolve('react-native-svg-transformer'),
  },
  resolver: {
    assetExts: assetExts.filter((ext) => ext !== 'svg'),
    sourceExts: [...sourceExts, 'cjs', 'mjs', 'svg'],
  },
};

module.exports = withNxMetro(
  withNativeWind(mergeConfig(defaultConfig, customConfig), {
    input: './global.css',
  }),
  {
    // Change this to true to see debugging info.
    // Useful if you have issues resolving modules
    debug: false,
    // all the file extensions used for imports other than 'ts', 'tsx', 'js', 'jsx', 'json'
    extensions: [],
    // Specify folders to watch, in addition to Nx defaults (workspace libraries and node_modules)
    watchFolders: [],
  },
);
```

**Expected behavior**
This should apply when starting the app

**Screenshots**

https://github.com/nativewind/nativewind/assets/42835486/e4a91ed8-b49b-43c3-ad85-4bb3a3938ee2

**Comment by nicholashamilton** [comment]
I'm having trouble getting nativewind v4 to work with NX + Expo Router too.

I created a minimal repo to display the issue.
https://github.com/nicholashamilton/nx-expo-nativewind-v4-issue

**Comment by danstepanov** [comment]
We are closing all pre-v4.1 issues now that v4.1 is released. If the problem is still happening with the latest v4.1, [please open a new issue](https://github.com/nativewind/nativewind/blob/next/contributing.md#opening-an-issue) and reference this one in the description.

**Comment by CristiCeban** [comment]
@nicholashamilton I've had exactly same issue as you, 
If you'll look here for [withNxMetro](https://github.com/nrwl/nx/blob/master/packages/expo/plugins/with-nx-metro.ts#L62)
It returns a Promise with an object which have `resolveRequest` key. This is a metro config object. If you're look at [withNativeWind](https://github.com/nativewind/nativewind/blob/main/packages/nativewind/src/metro/index.ts#L37), it uses internally [withCssInterpo](https://github.com/nativewind/nativewind/blob/main/packages/react-native-css-interop/src/metro/index.ts#L53), which do smart thing of memoizing resolveRequest function (withNxMetro is not doing this). It first call original [resolveRequest](https://github.com/nativewind/nativewind/blob/main/packages/react-native-css-interop/src/metro/index.ts#L186), and after that doing it's own logic. And it returns another Metro Config Object (not promise as WithNxMetro).
Now the correct approach here will be to call `withNxMetro` first, await for it, and after that pass config to the `withNativeWind`.

So you need to have something like that in your metro config
```js

const { withNxMetro } = require('@nx/expo');
const { getDefaultConfig } = require('@expo/metro-config');
const { mergeConfig } = require('metro-config');

const { withNativeWind } = require('nativewind/metro');

async function config(){
  const defaultConfig = getDefaultConfig(__dirname);
  const { assetExts, sourceExts } = defaultConfig.resolver;

  /**
   * Metro configuration
   * https://facebook.github.io/metro/docs/configuration
   *
   * @type {import('metro-config').MetroConfig}
   */
  const customConfig = {
    transformer: {
      babelTransformerPath: require.resolve('react-native-svg-transformer'),
    },
    resolver: {
      assetExts: assetExts.filter((ext) => ext !== 'svg'),
      sourceExts: [...sourceExts, 'cjs', 'mjs', 'svg'],
    },
  };
   
  const nxMetroConfig = await withNxMetro(mergeConfig(defaultConfig, customConfig),  {
    // Change this to true to see debugging info.
    // Useful if you have issues resolving modules
    debug: false,
    // all the file extensions used for imports other than 'ts', 'tsx', 'js', 'jsx', 'json'
    extensions: [],
    // Specify folders to watch, in addition to Nx defaults (workspace libraries and node_modules)
    watchFolders: [],
  });

  const finalConfig = withNativeWind(nxMetroConfig, { input: './global.css' });
  return finalConfig;
}

module.exports = config()
```

Now the question here is, why `withNxMetro` returns a promise, because from the current implementation it does all operation sync (reading stuff). But it's not really related to this thread.
Hope it will help you!

**Comment by lbxa** [comment]
@CristiCeban I've been wrestling with getting Nativewind to work with `react-native-svg-transformer` for the last 2 days and this was the last piece of the puzzle. Thank you!

**Comment by Seanmclem** [comment]
@danstepanov I still have this issue with 4.1, and I'm not using NX
